### PR TITLE
Extract potential host.domain and user.domain in Cisco Secure Endpoint

### PR DESCRIPTION
**Pull request by LaZyDK**
## What does this PR do?

Will try to extract host.domain, user.domain and user.email if possible. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by LaZyDK** [comment]
@P1llus please test :)

**Comment by P1llus** [comment]
I added some small additions to it, some extra failure checks in the scripts, ran the CI generation of test data, and ensured that both host fields are still set.

Just need another to review then I can merge.

File: packages/cisco_secure_endpoint/data_stream/event/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -177,18 +177,45 @@ processors:
...
+      def domain = "";
+      def nameArray = ctx.cisco.secure_endpoint.computer.hostname.toString().splitOnToken(".");
+      if (ctx.host == null) {
+        ctx.host = new HashMap();
+      }
+      if (nameArray?.length > 0) {
+        for (int i = 1; i < nameArray.length; i++) {
+          domain += nameArray[i] + (i < nameArray.length - 1 ? "." : "");
+        }
+        ctx.host.hostname = nameArray[0];
```
Comment (efd6) [inline packages/cisco_secure_endpoint/data_stream/event/elasticsearch/ingest_pipeline/default.yml:193]: Check that this is not ""? There are `host.domain: ""` showing up regularly in the outputs.

**Review by efd6 — COMMENTED** [review]
Remove `event.ingested` at https://github.com/elastic/integrations/pull/2888/files#diff-f430f71598ebb4c703c61a0500ebad83a87feee76e6f33c9ec8720576574e01eR38-R40?

None of the `host.domain` fields have any value other than "".

**Review by efd6 — APPROVED** [review]
LGTM

**Pull request merged**